### PR TITLE
Deprecate fslist1 channel

### PR DESCRIPTION
**Pull request by martinpitt**
Part of #19999

### CODE REVIEW

File: pkg/lib/cockpit-components-file-autocomplete.jsx [hunk 1]
```diff
@@ -91,38 +93,26 @@ export class FileAutoComplete extends React.Component {
 
     onCreateOption(newValue) {
         this.setState(prevState => ({
-            displayFiles: [...prevState.displayFiles, { type: "file", path: newValue }]
+            displayFiles: [...prevState.displayFiles, { type: "reg", path: newValue }]
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:96]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx [hunk 2]
```diff
@@ -91,38 +93,26 @@ export class FileAutoComplete extends React.Component {
...
-                (!this.props.onlyDirectories || item.type == 'directory')) {
-                item.path = item.path + (item.type == 'directory' ? '/' : '');
-                results.push(item);
-            }
-        });
+        fsinfo(path, ['type', 'entries'], { superuser: this.props.superuser })
+                .then(info => {
+                    console.log("updateFiles", path, "=>", info);
+                    const results = [];
+                    for (const name in info.entries ?? {}) {
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:108]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx [hunk 3]
```diff
@@ -91,38 +93,26 @@ export class FileAutoComplete extends React.Component {
...
+                        const type = info.entries[name].type;
+                        if (!this.props.onlyDirectories || type == 'dir')
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:110]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:96]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:108]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx (hunk 3 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:110]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:96]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:108]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx (hunk 3 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:110]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:96]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:108]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx (hunk 3 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:110]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:96]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:108]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx (hunk 3 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:110]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:96]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:108]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx (hunk 3 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:110]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:96]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:108]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx (hunk 3 above)
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:110]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx [hunk 4]
```diff
@@ -91,38 +93,25 @@ export class FileAutoComplete extends React.Component {
 
     onCreateOption(newValue) {
         this.setState(prevState => ({
-            displayFiles: [...prevState.displayFiles, { type: "file", path: newValue }]
+            displayFiles: [...prevState.displayFiles, { type: "reg", path: newValue }]
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:96]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx [hunk 5]
```diff
@@ -91,38 +93,25 @@ export class FileAutoComplete extends React.Component {
...
-            if (item && item.path && item.event == 'present' &&
-                (!this.props.onlyDirectories || item.type == 'directory')) {
-                item.path = item.path + (item.type == 'directory' ? '/' : '');
-                results.push(item);
-            }
-        });
+        fsinfo(path, ['type', 'entries'], { superuser: this.props.superuser })
+                .then(info => {
+                    const results = [];
+                    for (const name in info.entries ?? {}) {
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:107]: This added line is not executed by any test.

File: pkg/lib/cockpit-components-file-autocomplete.jsx [hunk 6]
```diff
@@ -91,38 +93,25 @@ export class FileAutoComplete extends React.Component {
...
+                        const type = info.entries[name].type;
+                        if (!this.props.onlyDirectories || type == 'dir')
```
Comment (cockpituous) [inline pkg/lib/cockpit-components-file-autocomplete.jsx:109]: This added line is not executed by any test.

File: doc/protocol.md [hunk 7]
```diff
@@ -1038,36 +1038,6 @@ fields:
  * "type": If the event was created this contains the type of the new file.
    Will be one of: file, directory, link, special or unknown.
 
-In case of an error, the channel will be closed.  In addition to the
```
Comment (mvollmer) [inline doc/protocol.md:1041]: This paragraph belongs to fswatch1. Shouldn't we keep it?

**Review by mvollmer — APPROVED** [review]
Thanks!

**Pull request merged**